src/PrestaShopBundle/Resources/config/services/adapter/order.yml

```diff
@@ -74,6 +74,8 @@ services:
 
   prestashop.adapter.order.command_handler.add_payment_handler:
     class: PrestaShop\PrestaShop\Adapter\Order\CommandHandler\AddPaymentHandler
+    arguments:
+      - '@=service("prestashop.adapter.legacy.context").getContext().employee.id'
```

Comment:
Injecting an Employee ID in a constructor feels very weird 🤔 

Dependency injection is most of the time used to inject services into services or services into controllers, not entity (or entity properties) into services. Services should be [stateless](https://www.redhat.com/en/topics/cloud-native-apps/stateful-vs-stateless) as much as possible. Stateful services introduce side-effects, for example if you use this service in CLI 😞 

If a users wants to pay an order, I expect something like
```
$paymentHandler->pay($order, $user)
```
not
```
$paymentHandler = new Handler($user);
$paymentHandler->pay($order);
```

Reply:
Hi, Could you explain me why it uses at many places ? 
![image](https://user-images.githubusercontent.com/1533248/178427497-7f673c92-7c36-4a80-bfe6-f08ce4c55e61.png)

Reply:
I missed that in my previous review...  but I think the employee ID should be put in the command, not in the command handler, then you can do something like `$command->getEmployeeId()->getValue()` instead of `$this->idEmployee` 👍

Reply:
you can find an exemple of this in `PrestaShop\PrestaShop\Core\Domain\Order\Command\AddOrderFromBackOfficeCommand` (see the constructor and the `getEmployeeId()` method